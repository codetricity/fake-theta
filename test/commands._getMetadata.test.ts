import {
  request,
  toMatchInvalidHeaderParameterError,
  toMatchMissingParameterError,
} from './helper';

describe('POST /osc/commands/execute _getMetadata', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMetadata',
        parameters: {
          fileUrl: 'http://192.168.1.1/files/100RICOH/R0010015.JPG',
        },
      })
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMetadata',
        parameters: { fileUri: 'dummy' },
      })
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA X POST /osc/commands/execute _getMetadata', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMetadata',
        parameters: {
          fileUrl: 'http://192.168.1.1/files/100RICOH/R0010015.JPG',
        },
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMetadata',
        parameters: { fileUri: 'dummy' },
      })
      .set('emulating-theta-model', 'x')
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA Z1 POST /osc/commands/execute _getMetadata', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMetadata',
        parameters: {
          fileUrl:
            'http://192.168.1.1/files/150100525831424d42075b53ce68c300/100RICOH/R0010015.JPG',
        },
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMetadata',
        parameters: { fileUri: 'dummy' },
      })
      .set('emulating-theta-model', 'z1')
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('Unknown Type POST /osc/commands/execute _getMySetting', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getMetadata',
        parameters: { fileUrl: 'dummy' },
      })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
