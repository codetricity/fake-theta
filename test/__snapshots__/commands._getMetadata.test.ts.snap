// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute _getMetadata should respond with results 1`] = `
{
  "exif": {
    "ApertureValue": 0,
    "BrightnessValue": 0,
    "ColorSpace": 1,
    "Compression": 6,
    "Copyright": "",
    "DateTime": "2015:07:10 11:05:18",
    "ExifVersion": "0231",
    "ExposureBiasValue": 0,
    "ExposureProgram": 2,
    "ExposureTime": 0.03333333333333333,
    "FNumber": 2.4,
    "Flash": 32,
    "FocalLength": 1.37,
    "GPSLatitude": 35.68045,
    "GPSLatitudeRef": "N",
    "GPSLongitude": 139.76905555555558,
    "GPSLongitudeRef": "E",
    "ImageDescription": "",
    "ImageLength": 5504,
    "ImageWidth": 11008,
    "Make": "RICOH",
    "Model": "RICOH THETA X ",
    "Orientation": 1,
    "PhotographicSensitivity": 320,
    "Software": "RICOH THETA X Ver 1.41.0",
    "WhiteBalance": 0,
  },
  "xmp": {
    "CroppedAreaImageHeightPixels": 5504,
    "CroppedAreaImageWidthPixels": 11008,
    "CroppedAreaLeftPixels": 0,
    "CroppedAreaTopPixels": 0,
    "FullPanoHeightPixels": 5504,
    "FullPanoWidthPixels": 11008,
    "ProjectionType": "equirectangular",
    "UsePanoramaViewer": true,
  },
}
`;

exports[`THETA X POST /osc/commands/execute _getMetadata should respond with results 1`] = `
{
  "exif": {
    "ApertureValue": 0,
    "BrightnessValue": 0,
    "ColorSpace": 1,
    "Compression": 6,
    "Copyright": "",
    "DateTime": "2015:07:10 11:05:18",
    "ExifVersion": "0231",
    "ExposureBiasValue": 0,
    "ExposureProgram": 2,
    "ExposureTime": 0.03333333333333333,
    "FNumber": 2.4,
    "Flash": 32,
    "FocalLength": 1.37,
    "GPSLatitude": 35.68045,
    "GPSLatitudeRef": "N",
    "GPSLongitude": 139.76905555555558,
    "GPSLongitudeRef": "E",
    "ImageDescription": "",
    "ImageLength": 5504,
    "ImageWidth": 11008,
    "Make": "RICOH",
    "Model": "RICOH THETA X ",
    "Orientation": 1,
    "PhotographicSensitivity": 320,
    "Software": "RICOH THETA X Ver 1.41.0",
    "WhiteBalance": 0,
  },
  "xmp": {
    "CroppedAreaImageHeightPixels": 5504,
    "CroppedAreaImageWidthPixels": 11008,
    "CroppedAreaLeftPixels": 0,
    "CroppedAreaTopPixels": 0,
    "FullPanoHeightPixels": 5504,
    "FullPanoWidthPixels": 11008,
    "ProjectionType": "equirectangular",
    "UsePanoramaViewer": true,
  },
}
`;

exports[`THETA Z1 POST /osc/commands/execute _getMetadata should respond with results 1`] = `
{
  "exif": {
    "ApertureValue": 4.97,
    "BrightnessValue": 6.8,
    "ColorSpace": 1,
    "Compression": 6,
    "Copyright": "",
    "DateTime": "2015:07:10 11:05:18",
    "ExifVersion": "0231",
    "ExposureBiasValue": 0,
    "ExposureProgram": 2,
    "ExposureTime": 0.0125,
    "FNumber": 5.6,
    "Flash": 32,
    "FocalLength": 2.57,
    "GPSLatitude": 35.68045,
    "GPSLatitudeRef": "N",
    "GPSLongitude": 139.76906,
    "GPSLongitudeRef": "E",
    "ImageDescription": "                                                               ",
    "ImageLength": 3360,
    "ImageWidth": 6720,
    "Make": "RICOH",
    "Model": "RICOH THETA Z1",
    "Orientation": 1,
    "PhotographicSensitivity": 125,
    "Software": "RICOH THETA Z1 Ver 2.20.3",
    "WhiteBalance": 0,
  },
  "xmp": {
    "CroppedAreaImageHeightPixels": 3360,
    "CroppedAreaImageWidthPixels": 6720,
    "CroppedAreaLeftPixels": 0,
    "CroppedAreaTopPixels": 0,
    "FullPanoHeightPixels": 3360,
    "FullPanoWidthPixels": 6720,
    "ProjectionType": "equirectangular",
    "UsePanoramaViewer": true,
  },
}
`;
